src/inbox.rs

```diff
@@ -19,7 +19,8 @@ use std::sync::{atomic, Arc, Mutex, Weak};
 
 type Spinlock<T> = spin::Mutex<T>;
 type MessageToOneActor<A> = Box<dyn MessageEnvelope<Actor = A>>;
-type BroadcastQueue<A> = Spinlock<BinaryHeap<MessageToAllActors<A>>>;
+type MessageToAllActors<A> = Box<dyn BroadcastEnvelope<Actor = A>>;
+type BroadcastQueue<A> = Spinlock<BinaryHeap<Prioritized<MessageToAllActors<A>>>>;
```

Comment:
Priorities are also stored 8 times

Reply:
So, I guess the question is primarily of whether this cost is worth the simplification gained

Reply:
Ah, yeah it makes total sense to use an `Arc` then.

Reply:
Am I mistaken or isn't the entire `BroadcastQueue` Arc'd anyway so there is only ever one actual instance?

Reply:
The reason it's `Arc`ed is to share it between the senders and the receivers, essentially. Although they could just be stored in the same allocation, sharing them separately with Arcs and Spinlocks means that they can be received from (after wakeups) without having to lock the entire channel again (which has much higher contention than a single broadcast channel). So, each receiver has one queue to itself. Maybe `BroadcastMailbox` would be a better name, actually, since this better communicates that there is one per receiver

Reply:
Okay, thanks for the explanation!

I am in favor of non-linear scaling with message size but it is an optimization that is worth documenting :)

One more question: What is the rationale behind `deep_clone` vs `shallow_weak_clone` on the receiver?
The production code seems to use `deep_clone` and the test uses `shallow_weak_clone`. Is that on purpose? `shallow_weak_clone` is not used anywhere else and it is the only place where `broadcast_queues` is modified. Something doesn't seem right here :)

Reply:
> The main difference in this change, each broadcast queue will store a full copy of each message - e.g, if it is 64 bytes on heap and there are 8 actors, sending a message will cost 64 * 8 on heap, versus 16 (counters) + 64 for the arc solution (ignoring vtables)

To summarise: With the current implementation, only the actual message is cloned which may not involve any additionally allocations depending on the particular message (i.e. it only consists of integers for example) whereas here we would definitely need an allocation per actor we dispatch the message to.

Reply:
I've done so, although the new stopgap solution to a problem I found when fixing the issue you raised is maybe less than perfect? I will think about it a bit more and let me know if you have a solution